pkg/scalers/kafka_scaler.go

```diff
@@ -364,26 +360,57 @@ func (s *kafkaScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricS
 	return []v2beta2.MetricSpec{metricSpec}
 }
 
+type consumerOffsetResult struct {
+	consumerOffsets *sarama.OffsetFetchResponse
+	err             error
+}
+
+type producerOffsetResult struct {
+	producerOffsets map[int32]int64
+	err             error
+}
+
+func (s *kafkaScaler) getConsumerAndProducerOffsets(partitions []int32) (*sarama.OffsetFetchResponse, map[int32]int64, error) {
+	consumerChan := make(chan consumerOffsetResult, 1)
```

Comment:
Does this need to be wrapped in a waitgroup.wait()?

The channels are never explicitly closed here: When you save the channel contents to the producerChan variable, how do we know that all producer offsets have actually been acquired?

I think* there needs to be a synchronization before you proceed with saving the channel contents.


*not a golang concurrency expert.

Reply:
No need here as receivers block if the buffered channel is empty. All producer offsets are sent in a single go back to the getConsumerAndProducersOffsets result chan.